spectral_cube/lower_dimensional_structures.py

```diff
@@ -246,6 +267,149 @@ def _quicklook_mpl(self, filename=None):
         if filename is not None:
             self.figure.savefig(filename)
 
+    def convolve_to(self, beam, convolve=convolution.convolve_fft):
+        """
+        Convolve each channel in the cube to a specified beam
+
+        Parameters
+        ----------
+        beam : `radio_beam.Beam`
+            The beam to convolve to
+        convolve : function
+            The astropy convolution function to use, either
+            `astropy.convolution.convolve` or
+            `astropy.convolution.convolve_fft`
+
+        Returns
+        -------
+        proj : `Projection`
+            A Projection convolved to the given ``beam`` object.
+        """
+
+        self._raise_wcs_no_celestial()
+
+        if not "beam" in self.meta:
+            raise ValueError("No beam is contained in Projection.meta.")
+
+        pixscale = wcs.utils.proj_plane_pixel_area(self.wcs.celestial)**0.5 * u.deg
+
+        convolution_kernel = \
+            beam.deconvolve(self.beam).as_kernel(pixscale)
+
+        newdata = convolve(self.value, convolution_kernel)
+
+        self = Projection(newdata, unit=self.unit, wcs=self.wcs,
+                          meta=self.meta, header=self.header,
+                          beam=beam)
+
+        return self
+
+    def reproject(self, header, order='bilinear'):
+        """
+        Reproject the image into a new header.
+
+        Parameters
+        ----------
+        header : `astropy.io.fits.Header`
+            A header specifying a cube in valid WCS
+        order : int or str, optional
+            The order of the interpolation (if ``mode`` is set to
+            ``'interpolation'``). This can be either one of the following
+            strings:
+
+                * 'nearest-neighbor'
+                * 'bilinear'
+                * 'biquadratic'
+                * 'bicubic'
+
+            or an integer. A value of ``0`` indicates nearest neighbor
+            interpolation.
+        """
+
+        self._raise_wcs_no_celestial()
+
+        try:
+            from reproject.version import version
+        except ImportError:
+            raise ImportError("Requires the reproject package to be"
+                              " installed.")
+
+        # Need version > 0.2 to work with cubes
```

Comment:
I think so. We could just remove it here, or keep it so the requirements are homogeneous.